src/encoder.rs

```diff
@@ -344,6 +346,56 @@ impl<T: Pixel> FrameState<T> {
     }
   }
 
+  pub fn compute_activity_mask(&mut self) {
+    //activity mask is computed only for luma plane
+
+    let luma_plane = &self.input.planes[0];
+    let PlaneConfig { width, height, .. } = luma_plane.cfg;
+    let luma_stride = luma_plane.cfg.stride;
+    let luma = &luma_plane.data;
+    let PlaneConfig { xdec, ydec, .. } = self.activity_mask.cfg;
+    assert!(xdec==3);
+    assert!(ydec==3);
+    let mut act_mask = vec![0 as u32; (width >> xdec) * (height >> ydec) ];
+
+    for r in 0..(height >> ydec) {
+      for c in 0..(width >> xdec) {
+        let base = luma_plane.index(r << xdec, c << ydec);
```

Comment:
The identifiers `r` and `c` are transposed here. The failing test case is resolved by the following:
```suggestion
        let base = luma_plane.index(c << xdec, r << ydec);
```

Reply:
Changed it.